File: solutions/0523-continuous-subarray-sum/solution.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> m;
        m[0]=-1;
int sum=0;
int rem;
        for(int i=0;i<nums.size();i++){
            sum=sum+nums[i];
            rem=sum%k;
            if(m.count(rem)){
                if(i-m[rem]>=2)
            return true;
            }
            else
            m[rem]=i;
            
            
        }
            return false;
    }
};

File: solutions/0523-continuous-subarray-sum/solution.py
class Solution:
    def checkSubarraySum(self, nums, k):
        mp = {0: -1}
        prefix = 0
        for i, num in enumerate(nums):
            prefix += num
            rem = prefix % k
            if rem in mp:
                if i - mp[rem] >= 2:
                    return True
            else:
                mp[rem] = i
        return False
